mibidata/tiff.py

```diff
@@ -82,12 +84,17 @@ def write(filename, image, sed=None, optical=None, ranges=None,
         raise ValueError('Image coordinates and size must not be None.')
     if image.masses is None or image.targets is None:
         raise ValueError('Image channels must contain both masses and targets.')
-    if np.issubdtype(image.data.dtype, np.integer) and not write_float:
-        range_dtype = 'I'
+    if write_dtype and not write_dtype in ['float', 'int', np.float, np.uint16]:
```

Comment:
Because the only int type we support is uint16, we cannot only check if an int subtype. I think if the user requests uint16 we should be checking if it can be losslessly converted and warning if not (an easy way to do this is to convert and then check equality with `np.all(np.equal(...))`).